src/screens/Signup/StepHandle.tsx

```diff
@@ -0,0 +1,123 @@
+import {useLingui} from '@lingui/react'
+import {msg, Trans} from '@lingui/macro'
+import {TimesLarge_Stroke2_Corner0_Rounded as Times} from '#/components/icons/Times'
+import {Check_Stroke2_Corner0_Rounded as Check} from '#/components/icons/Check'
+import {At_Stroke2_Corner0_Rounded as At} from '#/components/icons/At'
+import * as TextField from '#/components/forms/TextField'
+import {useSignupContext} from '#/screens/Signup/state'
+import {Text} from '#/components/Typography'
+import {atoms as a, useTheme} from '#/alf'
+import {
+  createFullHandle,
+  IsValidHandle,
+  validateHandle,
+} from 'lib/strings/handles'
+
+export function StepHandle() {
+  const {_} = useLingui()
+  const t = useTheme()
+  const {state, dispatch} = useSignupContext()
+
+  const [validCheck, setValidCheck] = React.useState<IsValidHandle>({
+    handleChars: false,
+    frontLength: false,
+    totalLength: true,
+    overall: false,
+  })
+
+  useFocusEffect(
+    React.useCallback(() => {
+      setValidCheck(validateHandle(state.handle, state.userDomain))
+      // Disabling this, because we only want to run this when we focus the screen
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []),
+  )
+
+  const onHandleChange = React.useCallback(
+    (value: string) => {
+      if (state.error) {
+        dispatch({type: 'setError', value: ''})
+      }
+
+      setValidCheck(validateHandle(value, state.userDomain))
+      dispatch({
+        type: 'setHandle',
+        value,
+      })
+    },
+    [dispatch, state.error, state.userDomain],
+  )
+
+  return (
+    <View style={[a.gap_xl]}>
+      <View>
+        <TextField.Root>
+          <TextField.Icon icon={At} />
+          <TextField.Input
+            onChangeText={onHandleChange}
+            label={_(msg`Input your user handle`)}
+            defaultValue={state.handle}
+            autoCapitalize="none"
+            autoCorrect={false}
+            autoFocus
+            autoComplete="off"
+          />
+        </TextField.Root>
+      </View>
+      <Text style={[a.text_md]}>
+        <Trans>Your full handle will be</Trans>{' '}
+        <Text style={[a.text_md, a.font_bold]}>
+          @{createFullHandle(state.handle, state.userDomain)}
+        </Text>
+      </Text>
+
+      <View
+        style={[
+          a.w_full,
+          a.rounded_sm,
+          a.border,
+          a.p_md,
+          a.gap_sm,
+          t.atoms.border_contrast_low,
+        ]}>
+        {state.error ? (
+          <View style={[a.w_full, a.flex_row, a.align_center, a.gap_sm]}>
+            <IsValidIcon valid={false} />
+            <Text style={[a.text_md, a.flex]}>{state.error}</Text>
+          </View>
+        ) : undefined}
+        <View style={[a.w_full, a.flex_row, a.align_center, a.gap_sm]}>
+          <IsValidIcon valid={validCheck.handleChars} />
+          <Text style={[a.text_md, a.flex]}>
+            <Trans>May only contain letters and numbers</Trans>
```

Comment:
Could this string be updated please to clarify that hyphens are also allowed in handles? (albeit not at the start or end, but I don't think that needs to be mentioned)

```suggestion
            <Trans>May only contain letters, numbers and hyphens</Trans>
```

Reply:
Agree it doesn't need to be shown usually, but still want to clarify if a user does put a `-` at the start or end. Added some logic to account for that and only display the message if thats the case. Thank you!

Reply:
That is a much better solution, actually 💫

Also, I think the new strings are clearer 👍  (I forgot the Oxford comma in en-US in my suggested version, oops 😅)

Reply:
Yea the other ones felt TOO polite haha, who says _MAY_? 😆